src/Hl7.Fhir.Shims.Base/Specification/Terminology/CodeSystemFilterProcessor.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Specification.Source;
+using Hl7.Fhir.Utility;
+using System.Collections.Generic;
+using System.Linq;
+using T = System.Threading.Tasks;
+
+#nullable enable
+
+namespace Hl7.Fhir.Specification.Terminology
+{
+
+    internal static class CodeSystemFilterProcessor
+    {
+        private const string SUBSUMEDBYCODE = "subsumedBy";
+
+        /// <summary>
+        /// Retrieve codes from a CodeSystem resource bases on one or multiple filters.
+        /// </summary>
+        /// <param name="codeSystemUri">Uri of the CodeSystem</param>
+        /// <param name="filters">Filters to be applied</param>
+        /// <param name="settings">ValueSetExpanderSettings </param>
+        /// <returns></returns>
+        /// <exception cref="ValueSetExpansionTooComplexException">Thrown when a filter is applied that is not supported (yet)</exception>
+        /// <exception cref="CodeSystemUnknownException">Thrown when no resource resolver was set in ValueSetExpanderSettings.ValueSetSource</exception>
+        /// <exception cref="CodeSystemUnknownException">Thrown when the requested CodeSystem can not be found by the resource resolver in ValueSetExpanderSettings</exception>
+        internal async static T.Task<IEnumerable<ValueSet.ContainsComponent>> FilterConceptsFromCodeSystem(string codeSystemUri, List<ValueSet.FilterComponent> filters, ValueSetExpanderSettings settings)
+        {
+            if (codeSystemUri == "http://snomed.info/sct" || codeSystemUri == "http://loinc.org")
```

Comment:
Can we make this configurable through the ValueSetExpanderSettings?

Reply:
The whole thing? or the list of valuesets you don't want to filter?

Reply:
The list of CodeSystem that should be reported as too complex

Reply:
added it